src/org/opensolaris/opengrok/search/context/ContextFormatter.java

```diff
@@ -0,0 +1,404 @@
+                            marks.add(line.substring(phi.getLineStart(),
+                                phi.getLineEnd()));
+                        }
+
+                        if (phi.getLineStart() < 0) {
+                            if (!didBold) bld.append(HtmlConsts.B);
+                            if (phi.getLineEnd() != Integer.MAX_VALUE) {
+                                lhi.hsub(bld, line, loff, phi.getLineEnd());
+                                loff += phi.getLineEnd() - loff;
+                            } else {
+                                lhi.hsub(bld, line, loff);
+                                loff = line.length();
+                            }
+                            bld.append(HtmlConsts.ZB);
+                            didBold = false;
+                        } else {
+                            lhi.hsub(bld, line, loff, phi.getLineStart());
+                            loff += phi.getLineStart() - loff;
+                            if (! didBold) {
+                                bld.append(HtmlConsts.B);
+                                didBold = true;
+                            }
+                        }
+                        if (phi.getLineEnd() == Integer.MAX_VALUE) {
+                            lhi.hsub(bld, line, loff);
+                            loff = line.length();
+                        } else {
+                            lhi.hsub(bld, line, loff, phi.getLineEnd());
+                            loff += phi.getLineEnd() - loff;
+                            if (didBold) {
+                                bld.append(HtmlConsts.ZB);
+                                didBold = false;
+                            }
+                        }
+                    }
+                }
+
+                if (didBold) {
+                    bld.append(HtmlConsts.ZB);
+                    // no need to unset didBold
+                }
+                finishLine(bld, lhi.getLineno(), marks);
+                // regardless of true EOF, write a <br/>
+                bld.append(HtmlConsts.BR);
+                /**
+                 * Appending a LF here would hurt the more.jsp view, while
+                 * search.jsp (where getContext() does it) is indifferent -- so
+                 * skip it.
+                 */
+                res.put(lhi.getLineno(), bld.toString());
+                bld.setLength(0);
+            } catch (IOException e) {
+                LOGGER.log(Level.SEVERE, "Could not format()", e);
+                return res;
+            }
+        }
+
+        res.setLimited(limited);
+        if (moreUrl != null) {
+            bld.append("<a href=\"");
+            bld.append(moreUrl);
+            bld.append("\">");
+            bld.append(MORE_LABEL);
+            bld.append("</a>");
+            bld.append(HtmlConsts.BR);
+            bld.append("\n");
+            res.setFooter(bld.toString());
+            bld.setLength(0);
```

Comment:
`bld` is local variable so it is eligible for garbage collection so this line is redundant, right? :)